core/src/main/java/org/apache/brooklyn/util/core/task/ValueResolver.java

```diff
@@ -203,6 +205,11 @@ public ResolverBuilderPretype(Object v) {
         this.swallowExceptions = true;
         return this;
     }
+
+    public ValueResolver<T> avoidSideEffects() {
```

Comment:
We should write a unit test for `ValueResolver`, as well as testing it in the yaml files.